docs/User-guide.rst

```diff
@@ -0,0 +1,73 @@
+**********
+User Guide
+**********
+
+Once you understand :ref:`how dedupe works <how-it-works-label>`, and you've taken
+a look at some of the :doc:`Examples`,
+you probably want to figure out how to adapt dedupe to your own problem. This
+user guide is for that.
+
+Memory Considerations
+=====================
+
+The most common memory bottleneck is during clustering. Within every block,
+after computing a score for each pair of records, dedupe then runs a
+connected-components algorithm as a pre-processing step for the hierarchical
+agglomerative clustering.
+`Currently, this requires having the entire edge list in memory
```

Comment:
this is not right, the edgelist stays as memmapped array, and mainly on disk. we do need to store what are basically pointers (array indices) to edges in memory but that can be a lot smaller than the edgelist (this is the components dict in the union_find algorithm).  

still the compent dict can take a lot of memory

Reply:
OK, makes sense. Sorry, I was just reading past issues and PRs and this was my best summary, but admittedly I wasn't confident in my understanding.

What do you think *are* the main memory bottlenecks then? Someone comes here after their machine runs out of memory, what are the first things to check or try? Thanks!

Reply:
i think the index predicates and the components dicts are probably the most likely places, in that order.